### ISSUE TITLE
Encoding-related error in gsub?

### ISSUE THREAD

**Issue by kogreger** [rstudio/shiny#976]
Hi!

I'm trying to run a Shiny app, which I developed on a Windows system with a German system locale on an RHEL 7 server now, but the app crashes immediately after startup. I tried to step through it using RStudio Server von the Linux machine to track down the error and included `options(shiny.trace = TRUE)` before starting up the app.

This is the last message of the console output:
`Error in gsub("(?m)base64,[a-zA-Z0-9+/=]+", "[base64 data]", json, perl = TRUE) : input string 1 is invalid UTF-8`

Digging through the issues here in the repository I came across https://github.com/rstudio/shiny/issues/742, which was related to an encoding issue in `grepl`. Like in that case the system locale of my RHEL server is also German, as can be seen from the `sessionInfo()`:

```R
R version 3.2.2 (2015-08-14)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

locale:
 [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C               LC_TIME=de_DE.UTF-8        LC_COLLATE=de_DE.UTF-8     LC_MONETARY=de_DE.UTF-8   
 [6] LC_MESSAGES=de_DE.UTF-8    LC_PAPER=de_DE.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RPostgreSQL_0.4    DBI_0.3.1          stringr_1.0.0      shiny_0.12.2       rgeos_0.3-13       rgdal_1.0-7        sp_1.2-0          
 [8] reshape2_1.4.1     readr_0.1.1        RColorBrewer_1.1-2 leaflet_1.0.0      ggvis_0.4.2        dplyr_0.4.3       

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.1      magrittr_1.5     munsell_0.4.2    colorspace_1.2-6 xtable_1.7-4     lattice_0.20-33  R6_2.1.1         plyr_1.8.3      
 [9] tools_3.2.2      parallel_3.2.2   grid_3.2.2       htmltools_0.2.6  yaml_2.1.13      lazyeval_0.1.10  assertthat_0.1   digest_0.6.8    
[17] htmlwidgets_0.5  mime_0.4         stringi_0.5-5    scales_0.3.0     jsonlite_0.9.17  httpuv_1.3.3
```

Only strange thing is that the user in the other issue experienced the problem on Windows while it was hard to replicate on MacOS/Unix, while in my case it works perfectly fine on Windows but crashes on Linux.

I'd greatly appreciate any ideas.

**Comment by yihui** [comment]
Did you save everything in UTF-8 on Windows? (cf http://shiny.rstudio.com/articles/unicode.html)

**Comment by kogreger** [comment]
All the data is coming from a PostgreSQL database, whose `LC_COLLATE = 'de_DE.UTF-8'` and `LC_CTYPE = 'de_DE.UTF-8'`, so that shouldn't be an issue.

**Comment by yihui** [comment]
The fact that your app works on Windows may well be just a coincidence. You can try this branch and see if your app still works on Windows:

```r
devtools::install_github('yihui/shiny@bugfix/961-unicode')
```

BTW, when the app crashes on Linux, what exactly do you mean by "crash"? Does the screen gray out, or is there an error message in the R console? Are there any error messages in the JavaScript console of your web browser?

**Comment by wch** [comment]
Closing for lack of response. You can re-open or file a new issue if this comes up again.

**Comment by mimaque** [comment]
Hello, I am also experiencing this error on my Ubuntu 16.04 machine. Here is the output:
```
Warning: Error in gsub: input string 1 is invalid in this locale
Stack trace (innermost first):
    72: gsub
    71: dbQuoteIdentifier
    70: dbQuoteIdentifier
    69: .local
    68: dbWriteTable
    67: dbWriteTable
    66: write_df_as_table
    65: observeEventHandler [/srv/shiny-server/server.R#5258]
     1: shiny::runApp

```

**Comment by mimaque** [comment]
This seems to be working for me. I am running into this problem when launching shiny inside a docker container. Adding these lines to the image seems to remove the problem: http://jaredmarkell.com/docker-and-locales/
